src/kernelbot/cogs/admin_cog.py

```diff
@@ -512,6 +522,153 @@ async def start(self, interaction: discord.Interaction):
             interaction, "Bot will accept submissions again!", ephemeral=True
         )
 
+    def _parse_user_id_arg(self, user_id: str) -> str:
+        """Accepts a raw id or a discord mention and returns the id string."""
+        s = (user_id or "").strip()
+        if s.startswith("<@") and s.endswith(">"):
+            s = s[2:-1].strip()
+            if s.startswith("!"):
+                s = s[1:].strip()
+        return s
+
+    def _format_rate(self, rate: float | None) -> str:
+        if rate is None:
+            return "unlimited"
+        r = float(rate)
+        if r == 0:
+            return "blocked"
+        return f"{r:g}/min"
+
+    @app_commands.describe(
+        rate_per_minute="Rate in submissions/minute. Use 'none' for unlimited; 'default' clears a user override.",  # noqa: E501
+        user_id="Optional user id or mention. If omitted, sets the default.",
+    )
+    @with_error_handling
+    async def set_submission_rate_limit(
+        self,
+        interaction: discord.Interaction,
+        rate_per_minute: str,
+        user_id: Optional[str] = None,
+    ):
+        is_admin = await self.admin_check(interaction)
+        if not is_admin:
+            await send_discord_message(
+                interaction,
+                "You need to be Admin to use this command.",
+                ephemeral=True,
+            )
+            return
+
+        rate_s = (rate_per_minute or "").strip().lower()
```

Comment:
this code is too defensive, it's mainly us mods making the change right? in which case we don't need to be this paranaoid

Also now as a mod i'm not sure what value should i set closer to 1 or 0.5, it's partly why I favor unlimited or at most one submission just cause it's easier for me to intuit what to put in